src/fromager/resolver.py

```diff
@@ -200,6 +200,7 @@ def get_project_from_pypi(
     [str, RequirementsMap, CandidatesMap],
     typing.Iterable[tuple[str, str | Version]],
 ]
+VersionsMap: typing.TypeAlias = typing.Mapping[str, typing.Iterable[Version]]
```

Comment:
I think my explanation of VersionMap wasn't clear.

The idea was to have VersionMap be an association from a Version to an arbitrary value. Then the user of the map could query the map with a Requirement specifier and Constraint, and have the map give back the value associated with the version that matches that rule.

Our plugin could then instantiate a VersionMap, populate it, and query it without having to set up and use a Provider on its own.

I've put up an example in #542